src/software/amazon/ion/impl/lite/IonStructLite.java

```diff
@@ -402,9 +401,9 @@ public IonValue get(String fieldName)
         IonValue field;
 
         if (field_idx < 0) {
+            if(hasNullFieldName) throw new UnknownSymbolException("Because a field exists with an unknown name we cannot say for certain whether the requested field exists or not.");
```

Comment:
This condition needs a test.

Reply:
Also, the message is a bit wordy. The use of 'we', at least, feels too conversational for an exception message to me.

Reply:
agreed ill add one.